src/formatDate.js

```diff
@@ -50,7 +53,68 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const prefixYear = (yearBefore, newFormat) => {
+    const yearUpdateFormat
+      = newFormat.find(partDate => partDate.includes('YY'));
+
+    const previousCentury = '19';
+    const currentCentury = '20';
+    const toDetermineCentury = 30;
+
+    if (yearBefore.length === yearUpdateFormat.length) {
+      return yearBefore;
+    }
+
+    if (yearBefore.toString().length > yearUpdateFormat.length) {
+      return yearBefore.slice(-2);
+    }
+
+    if (yearUpdateFormat.length === 4 && yearBefore >= toDetermineCentury) {
+      return previousCentury + yearBefore;
+    }
+
+    if (yearUpdateFormat.length === 4 && yearBefore < toDetermineCentury) {
+      return currentCentury + yearBefore;
+    }
+
+    return yearBefore;
+  };
+
+  const newFormatDate = [];
+  const dateArr = date.split(fromFormat[fromFormat.length - 1]);
+  const indexYY = fromFormat
+    .findIndex((partDate) => partDate === 'YY' || partDate === 'YYYY');
+
+  const indexMM = fromFormat.indexOf('MM');
+  const indexDD = fromFormat.indexOf('DD');
+
+  const day = dateArr[indexDD];
+  const month = dateArr[indexMM];
+  const year = dateArr[indexYY];
+
+  for (let i = 0; i < toFormat.length; i++) {
+    const separator = toFormat[toFormat.length - 1];
+
+    if (toFormat[i] === 'DD') {
+      newFormatDate.push(day);
+      newFormatDate.push(separator);
+
+    }
+
+    if (toFormat[i] === 'MM') {
+      newFormatDate.push(month);
+      newFormatDate.push(separator);
+    }
+
+    if (toFormat[i] === 'YY' || toFormat[i] === 'YYYY') {
+      const yearUpdate = prefixYear(year, toFormat);
+
+      newFormatDate.push(yearUpdate);
+      newFormatDate.push(separator);
+    }
+  }
```

Comment:
Why do you add a `separator` each time separately? Wouldn't it be easier to collect all the values in an array and combine them using a `join` specifying a new `separator` as an argument?